### PR TITLE
[Feature] filter signals with missing data

### PR DESCRIPTION
**Pull request by danibene**
# Description
As a step towards better handling of missing data (see https://github.com/neuropsychology/NeuroKit/issues/838), this PR aims to allow for filtering signals even if they contain NaNs.

# Proposed Changes

1. I changed the `signal_filter()` function so that NaNs are interpolated before filtering and then added back after filtering.
2. I added `test_signal_filter_with_missing()` to test that the signal containing NaNs is filtered.

# Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/neuropsychology/NeuroKit/blob/master/.github/CONTRIBUTING.rst#structure-and-code) file.
- [x] My PR is targeted at the **dev branch** (and not towards the master branch).

### CODE REVIEW

**Comment by danibene** [comment]
I wonder, should we have a warning at this stage if missing samples are detected (or a certain proportion of missing samples)? Or only have warnings in specific feature extraction functions?

File: neurokit2/signal/signal_filter.py [hunk 1]
```diff
@@ -353,3 +358,12 @@ def _signal_filter_windowsize(window_size="default", sampling_rate=1000):
         if (window_size % 2) == 0:
             window_size + 1  # pylint: disable=W0104
     return window_size
+
+
+def _signal_filter_missing(signal):
+    """Interpolate missing data and save the indices of the missing data."""
+    missing = np.where(np.isnan(signal))[0]
+    if len(missing) > 0:
+        return signal_interpolate(signal), missing
```
Comment (DominiqueMakowski) [inline neurokit2/signal/signal_filter.py:367]: We should probably specify a linear / padding interpolation so that it's fast

**Comment by DominiqueMakowski** [comment]
> I wonder, should we have a warning at this stage if missing samples are detected (or a certain proportion of missing samples)? Or only have warnings in specific feature extraction functions?

Mmh good question... any preference?

**Comment by danibene** [comment]
> > I wonder, should we have a warning at this stage if missing samples are detected (or a certain proportion of missing samples)? Or only have warnings in specific feature extraction functions?
> 
> Mmh good question... any preference?


Not really, I noticed that there is already a warning in `ecg_clean()`:
https://github.com/neuropsychology/NeuroKit/blob/11708876e4a73d8653c90f342198492df692e1be/neurokit2/ecg/ecg_clean.py#L89-L96

So I guess to avoid being repetitive we can leave it out for now.

**Comment by DominiqueMakowski** [comment]
Sounds good, feel free to merge

**Pull request merged**